x/consensus/keeper/keeper.go

```diff
@@ -99,3 +104,31 @@ func (k Keeper) UpdateParams(ctx context.Context, msg *types.MsgUpdateParams) (*
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+func (k Keeper) SetCometInfo(ctx context.Context, msg *types.MsgSetCometInfo) (*types.MsgSetCometInfoResponse, error) {
+	if !bytes.Equal(coreapp.ConsensusIdentity, []byte(msg.Authority)) {
+		return nil, fmt.Errorf("invalid authority; expected %s, got %s", coreapp.ConsensusIdentity, msg.Authority)
+	}
+
+	cometInfo := types.CometInfo{
+		Evidence:        msg.Evidence,
+		ValidatorsHash:  msg.ValidatorsHash,
+		ProposerAddress: msg.ProposerAddress,
+		LastCommit:      msg.LastCommit,
+	}
+
+	if err := k.cometInfo.Set(ctx, cometInfo); err != nil {
+		return nil, err
+	}
+
+	return &types.MsgSetCometInfoResponse{}, nil
+}
+
+func (k Keeper) CometInfo(ctx context.Context, _ *types.QueryCometInfoRequest) (*types.QueryCometInfoResponse, error) {
```

Comment:
would love to rename it to `GetCometInfo`

Reply:
on it